src/agent/onefuzz/src/expand.rs

```diff
@@ -110,23 +110,14 @@ impl<'a> Expand<'a> {
             PlaceHolder::InputFileSha256.get_string(),
             ExpandedValue::Mapping(Box::new(Expand::input_file_sha256)),
         );
-        values.insert(
-            PlaceHolder::MachineId.get_string(),
-            ExpandedValue::Mapping(Box::new(Expand::machine_id)),
-        );
 
         Self { values }
     }
 
-    // Note: this spawns it's own tokio runtime as it's not really workable to
-    // refactor the get_machine_id into a blockable version, however... as this
-    // is implemented as a ExpandedValue::Mapping, launching a tokio runtime
-    // occurs when users actively attempt to expand {machine_id}.
-    fn machine_id(&self, _format_str: &str) -> Result<Option<ExpandedValue<'a>>> {
-        let rt = tokio::runtime::Runtime::new()?;
-        let machine_id = rt.block_on(get_machine_id())?;
-        let value = machine_id.to_hyphenated().to_string();
-        Ok(Some(ExpandedValue::Scalar(value)))
+    pub async fn machine_id(self) -> Result<Expand<'a>> {
```

Comment:
Let's add a comment to remind ourselves why this exists.
```suggestion
    // Must be manually called to enable the use of async library code.
    pub async fn machine_id(self) -> Result<Expand<'a>> {
```